cypress/support/reusable_tests/admin.js

```diff
@@ -0,0 +1,194 @@
+        expect($element).to.be.visible;
+        expect($element).to.have.attr('type','button');
+        expect($element).contain('New Manager');
+    });
+
+    cy.getElementByTestId('admin-users-title').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).contain('Data Managers');
+    });
+
+    cy.getElementByTestId('filter-users-title').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).contain('Filter Users');
+    });
+
+    cy.getElementByTestId('filter-users-type').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).contain('Type:');
+    });
+
+    cy.getElementByTestId('filter-users-search').then($element=>{
+        expect($element).to.be.visible;
+    });
+
+    cy.getElementByTestId('filter-users-search-button').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).to.have.attr('type','button');
+        expect($element).contain('Search');
+    });
+
+    cy.getElementByTestId('table').then($element=>{
+        expect($element).to.be.visible;
+    });
+
+    cy.getElementByTestId('table-header').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).to.have.lengthOf(2);
+    });
+
+    cy.getElementByTestId('table-header-text').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).to.have.lengthOf(2);
+        expect($element[0]).contain('Name');
+        expect($element[1]).contain('Email');
+    });
+
+    cy.getElementByTestId('table-row').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).to.have.length.greaterThan(1);
+    });
+
+    cy.getElementByTestId('pagination-previous').then($element=>{
+        expect($element).to.be.visible;
+    });
+
+    cy.getElementByTestId('pagination-next').then($element=>{
+        expect($element).to.be.visible;
+    });
+
+    cy.getElementByTestId('pagination-page').then($element=>{
+        expect($element).to.be.visible;
+    });
+    
+});
+
+Cypress.Commands.add('testAdminFilterUsers',(viewport,creds)=>{
+    cy.viewport(viewport);
+    cy.login(creds.email,creds.password);
+
+    cy.getElementByTestId('header-admin-link').click();
+    cy.getElementByTestId('filter-users-search').select('lawyer');
+    cy.getElementByTestId('filter-users-search-button').click();
+    //Should be empty
+    cy.getElementByTestId('admin-search-not-found-title').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).contain('No results found.');
+    });
+
+    cy.getElementByTestId('admin-search-not-found-body').then($element=>{
+        expect($element).to.be.visible;
+        expect($element).contain('Please refine your search');
+    });
+    
+    cy.getElementByTestId('filter-users-search').select('dataManager');
+    cy.getElementByTestId('filter-users-search-button').click();
+    cy.wait(500)
```

Comment:
@Alfredo-Moreira we can use `cy.intercept()` to automatically wait till the request has completed without introducing a manual wait. See [Cypress docs](https://docs.cypress.io/api/commands/intercept#Aliasing-an-intercepted-route) for more details